app/src/main/java/com/android/partagix/model/LoanViewModel.kt

```diff
@@ -45,14 +47,27 @@ class LoanViewModel(
       viewModelScope.launch {
         db.getAvailableItems(true) { newItems: List<Item> ->
           db.getUsers { users: List<User> ->
+            val newAvailabilityList = mutableListOf<Boolean>()
             val usersMap = users.associateBy { it.id }
             availableLoans =
                 newItems.map { item ->
                   val u = usersMap[item.idUser]!!
                   LoanDetails(item, u)
                 }
             applyFilters(uiState.value.filterState)
+            newItems.map { item ->
+              db.getItemUnavailability(item.id) { date ->
+                newAvailabilityList.add(
+                    if (date.none { stripTime(it) == stripTime(Date()) }) {
+                      true
+                    } else {
+                      false
+                    })
```

Comment:
For better readability:

```suggestion
                newAvailabilityList.add(date.none { stripTime(it) == stripTime(Date()) })
```